src/osm-gps-map/osm-gps-map-qt.cpp

```diff
@@ -219,6 +221,14 @@ Maep::GpsMap::GpsMap(QQuickItem *parent)
   bool orientation = gconf_get_bool(GCONF_KEY_SCREEN_ROTATE, TRUE);
   int gpsRefresh = gconf_get_int(GCONF_KEY_GPS_REFRESH_RATE, 1000);
   bool compassEnabled = gconf_get_bool(GCONF_KEY_COMPASS_ENABLED, FALSE);
+  gint compassMode = std::min(static_cast<gint>(COMPASS_N_MODES) - 1,
+                              std::max(static_cast<gint>(COMPASS_MODE_OFF),
+                                        gconf_get_int(GCONF_KEY_COMPASS_MODE,
+                                                      static_cast<gint>(COMPASS_MODE_OFF))));
+
+  // backward compatibility with old boolean setting
+  if (compassEnabled && compassMode == COMPASS_MODE_OFF)
+    compassMode = COMPASS_MODE_NORTH;
```

Comment:
Oh, I assumed that when saving the config I start with a "clean slate", so that when I don't save `GCONF_KEY_COMPASS_ENABLED` in the destructor it won't exist anymore on the next startup. So I guess what I want to do is delete that deprecated key from the dconf backend explicitly, or would you say that's unnecessary?